### PR TITLE
Add an error message for illegal start of simple pattern - Parsers.scala:1437

### PR DESCRIPTION
**Pull request by maseev**
This pull request is related to the following [issue](https://github.com/lampepfl/dotty/issues/1589).

### CODE REVIEW

File: src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 1]
```diff
@@ -789,4 +789,10 @@ object messages {
     val msg = "unreachable code"
     val explanation = ""
   }
+
+  case class IllegalStartOfSimplePattern()(implicit ctx: Context) extends Message(32) {
+    val kind = "Syntax"
+    val msg = "illegal start of simple pattern"
+    val explanation = ""
```
Comment (felixmulder) [inline src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: This message doesn't really add anything to justify its existence currently. Some things to consider for the explanation:

* When does this occur?
* What is a simple expression?
* Is there perhaps something that could be added to the message in order to help tooling / the user?

Let's try to put some semantics into this :)

**Comment by maseev** [comment]
@felixmulder PTAL

File: src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 2]
```diff
@@ -812,4 +812,83 @@ object messages {
...
+        """def swap(tuple: (String, Int)): (Int, String) = tuple match {
+          |  case (text, number) => (number, text)
+          |}
+        """
+      val patternSequencesCode =
+        """def getSecondValue(list: List[Int]): Int = list match {
+          |  case List(_, second, x:_*) => second
+          |  case _ => 0
+          |}"""
+      hl"""|Simple pattern can be divided into several groups:
```
Comment (felixmulder) [inline src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: patterns*

File: src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 3]
```diff
@@ -812,4 +812,83 @@ object messages {
...
+           |- Variable Patterns: ${"case x => ..."}.
+           |  It matches any value, and binds the variable name to that value.
+           |  A special case is the wild-card pattern _ which is treated as if it was a fresh
+           |  variable on each occurrence.
+           |
+           |- Typed Patterns: ${"case x:Int => ..."} or ${"case _:Int => ..."}.
```
Comment (felixmulder) [inline src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: space after `:`

**Comment by maseev** [comment]
@felixmulder PTAL. BTW, I ran the 'partest' task from 'sbt' and here's what I've got:

```
1802/1802 passed (elapsed time: 00:18:50)
Test Run PASSED
[success] Total time: 1134 s, completed Nov 15, 2016 6:52:46 PM
```
I'm not really sure why the jenkins build failed :disappointed:

**Comment by felixmulder** [comment]
Let's just try rebuilding on the CI first

**Comment by maseev** [comment]
it failed again :(

**Comment by felixmulder** [comment]
Hmm, yes. Your branch's working on my machine - @smarter, could you rerun this on a different behemoth?

**Comment by maseev** [comment]
@felixmulder Looks like it's working again! PTAL

**Comment by felixmulder** [comment]
LGTM, thanks @maseev!

**Pull request merged**